src/webview/VSCodeKaotoEditorChannelApi.ts

```diff
@@ -1,8 +1,20 @@
+import { KaotoEditorChannelApi } from '@kaoto/kaoto';
+import { ISettingsModel, SettingsModel } from '@kaoto/kaoto/models';
 import { DefaultVsCodeKieEditorChannelApiImpl } from '@kie-tools-core/vscode-extension/dist/DefaultVsCodeKieEditorChannelApiImpl';
 import * as vscode from 'vscode';
 
-export class VSCodeKaotoEditorChannelApi extends DefaultVsCodeKieEditorChannelApiImpl {
-    async getCatalogURL(): Promise<string | undefined> {
-        return await vscode.workspace.getConfiguration('kaoto').get('catalog.url');
-    }
+export class VSCodeKaotoEditorChannelApi
+  extends DefaultVsCodeKieEditorChannelApiImpl
+  implements KaotoEditorChannelApi
+{
+  async getCatalogURL(): Promise<string | undefined> {
+    return await vscode.workspace.getConfiguration('kaoto').get('catalog.url');
+  }
+
+  async getVSCodeKaotoSettings(): Promise<ISettingsModel> {
+    const catalogUrl =
+      vscode.workspace.getConfiguration('kaoto').get<string | null>('catalog.url') ?? '';
```

Comment:
from a style point of view, you can really write on the same line the1 07 characters. I think almost everybody have monitor  which can handle a lot more than 80 characters nowadays

Reply:
Sure, it probably picked the settings from Kaoto workspace, I'll make it one line